### ISSUE TITLE
Show neighbouring practices/CCGs on dashboard map

### ISSUE THREAD

**Issue by HelenCEBM** [ebmdatalab/openprescribing#394]
We could perhaps make the maps at the top of the practice/CCG dashboards more helpful if they showed other practices/CCGs in the vicinity (in a different colour), so users can easily explore behaviours at nearby locations. 

@sebbacon estimates 1+ days work (not having looked at the code)

**Comment by sebbacon** [comment]
This is probably more complicated than I thought, as we would need to alter the backend to return practices within a certain bounding box, and feeding these to the ratio calculations.

We would then need to update the practices and ratios as the bounding box changes (on a user drag event).

I think this is more likely to be 4 days' work and I wonder if the demand for this feature justifies the effort.

**Comment by HelenCEBM** [comment]
Not really sure what you mean by ratio calculations..? Rather than the ones nearby, is it possible to just map all the practices whether or not they are in view around the selected practice?

**Comment by sebbacon** [comment]
>  is it possible to just map all the practices whether or not they are in view around the selected practice?
Yes: "whether or not they are in view" == "within a certain bounding box"

> Not really sure what you mean by ratio calculations

We compute a ratio on the client side based on numerator/denominator pairs we get from the API. The API returns these pairs for all practices within a given CCG.

The change would be to:

* change the API to accept a bounding box rather than a CCG id
* make the map code request all practices within the initial bounding box (i.e. including those not in the CCG but on the visible part of the map)
* We should visually distinguish between practices in the CCG and others that are just visible. 

Adding practices that are not in the CCG would then create a UI expectation that dragging and zooming would add practices as the bounding box changed, so we would also have to:

*  set some kind of limit to the zoom so we don't request more practice data than is possible to service by the database in one go; and 
* change the way we call the API from the map code so it only appends things to the map, rather than re-gets all the data.

**Comment by sebbacon** [comment]
Having discussed with Helen, I realise I forgot what this was about while investigating it.

The context of the original idea is that it's hard to know when practices have closed / merged. One way of telling this is to plot the list size of a practice over time on its main dashboard page. Interpreting which practice has possibly taken over the list might be easier if you could navigate to other practice dashboards via a map.

Implementation would be similar to the above, just without ratio data. The trickiest bit is setting an expectation of being able to show new practices by panning or zooming the map. It might be useful enough only to show practices within a CCG, in which case we could remove that expectation and make implementation simpler.

**Comment by lmfrench** [comment]
I think we should wait until we've heard back from the BSA. They combine merged practices in ePACT, and if they will share this dataset with us there may be an easier way for us to show users when practices have merged (see #215)
